connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryDataSourceReader.java

```diff
@@ -85,11 +85,13 @@ public BigQueryDataSourceReader(
         new ReadSessionCreator(readSessionCreatorConfig, bigQueryClient, bigQueryReadClientFactory);
     this.globalFilter = globalFilter;
     this.schema = schema;
-    this.fields =
+    // We want to keep the key order
+    this.fields = new LinkedHashMap<>();
+    for (StructField field :
         JavaConversions.asJavaCollection(
```

Comment:
Java `Collection` doesn't guarantee ordering. Should we use a something like a list? (Using something like `JavaConversions .seqAsJavaList`)

Reply:
fixed